plugins/modules/web_infrastructure/jira.py

```diff
@@ -320,12 +371,33 @@ def edit(restbase, user, passwd, params):
     return ret
 
 
+def update(restbase, user, passwd, params):
+    data = {
+        "update": params['fields']
+    }
+    url = restbase + '/issue/' + params['issue']
+
+    ret = put(url, user, passwd, params['timeout'], data)
+
+    return ret
+
+
 def fetch(restbase, user, passwd, params):
     url = restbase + '/issue/' + params['issue']
     ret = get(url, user, passwd, params['timeout'])
     return ret
 
 
+def search(restbase, user, passwd, params):
+    url = restbase + '/search?jql=' + urllib.request.pathname2url(params['jql']) + '&maxResults=10'
```

Comment:
```suggestion
    url = restbase + '/search?jql=' + urllib.request.pathname2url(params['jql']) + '&maxResults=10'
```
why is maxResults exactly 10? maybe worth doing it configurable anyhow or, at least, mentioning this restriction in the documentation (if it really needs to be hardcoded as 10)?

Reply:
Maybe it should be the jira default and the parameter could be removed

Reply:
probably the default will be ok, i think

Reply:
I suggest adding maxresults as a module param with a default that matches the Jira default, firstly, to make it clear there is a limit to the number of records returned, and secondly so there's a way to tweak it if you need to.

Reply:
yep, to make it configurable sounds good

Reply:
> @pertoft
> 
> 1. what about @simonbaird 's suggestion?
> 2. we need a person who'll check the module with these changes manually and confirm that all parameters work ok
1)
I have updated the search module support a maxresults parameter for jira search:
`def search(restbase, user, passwd, params):
    url = restbase + '/search?jql=' + urllib.request.pathname2url(params['jql'])
    if params['fields']:
        fields = params['fields'].keys()
        url = url + '&fields=' + '&fields='.join([urllib.request.pathname2url(f) for f in fields])
    if params['maxresults']:
        url = url + '&maxResults=' + str(params['maxresults'])
    ret = get(url, user, passwd, params['timeout'])
    return ret
`
2) 
` - name: Get CMDB issue
     jira:
        uri: 'https://jira.netic.dk/'
        username: '{{ ansible_user }}'
        password: '{{ jira_password }}'
        project: CMDB
        operation: search
        maxresults: 2
        jql: "text ~ 'esx*'"
        validate_certs: no
     args:
        fields:
            customfield_10000:
     register: cmdb_issue
     delegate_to: localhost

   - debug:
      var: cmdb_issue`
results in:
`
PLAY [all] *****************************************************************************************************************************************************************************************************************************************************************************************************************
META: ran handlers
Read vars_file '~/infrastructure/ansible/vars/vault.yml'
Read vars_file '~/infrastructure/ansible/vars/vault.yml'

TASK [Get CMDB issue] ******************************************************************************************************************************************************************************************************************************************************************************************************
task path: /Users/pto/infrastructure/ansible/playbooks/jira/jira-cmdb-get.yml:11
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: pto
<localhost> EXEC /bin/sh -c 'echo ~pto && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077 `" && echo ansible-tmp-1586348301.741537-113582586070077="` echo /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077 `" ) && sleep 0'
Using module file /Users/pto/infrastructure/ansible/library/jira.py
<localhost> PUT /Users/pto/.ansible/tmp/ansible-local-82993isgaqx1b/tmp3nqzvtz7 TO /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077/AnsiballZ_jira.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077/ /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077/AnsiballZ_jira.py && sleep 0'
<localhost> EXEC /bin/sh -c '/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077/AnsiballZ_jira.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /Users/pto/.ansible/tmp/ansible-tmp-1586348301.741537-113582586070077/ > /dev/null 2>&1 && sleep 0'
changed: [dc4-1-b6-esx48.curasaas.netic.dk -> localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "assignee": null,
            "comment": null,
            "description": null,
            "fields": {
                "customfield_10000": null
            },
            "inwardissue": null,
            "issue": null,
            "issuetype": null,
            "jql": "text ~ 'esx*'",
            "linktype": null,
            "maxresults": 2,
            "operation": "search",
            "outwardissue": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "project": "CMDB",
            "status": null,
            "summary": null,
            "timeout": 10.0,
            "uri": "https://jira.netic.dk/",
            "username": "pto",
            "validate_certs": false
        }
    },
    "meta": {
        "expand": "schema,names",
        "issues": [
            {
                "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
                "fields": {
                    "customfield_10000": "netic.dk"
                },
                "id": "285131",
                "key": "x-6866",
                "self": "https://jira.netic.dk/rest/api/2/issue/285131"
            },
            {
                "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
                "fields": {
                    "customfield_10000": "netic.dk"
                },
                "id": "290184",
                "key": "x-592",
                "self": "https://jira.netic.dk/rest/api/2/issue/290184"
            }
        ],
        "maxResults": 2,
        "startAt": 0,
        "total": 2494
    }
}
Read vars_file '~/infrastructure/ansible/vars/vault.yml'

TASK [debug] ***************************************************************************************************************************************************************************************************************************************************************************************************************
task path: /Users/pto/infrastructure/ansible/playbooks/jira/jira-cmdb-get.yml:27
ok: [netic.dk] => {
    "cmdb_issue": {
        "changed": true,
        "failed": false,
        "meta": {
            "expand": "schema,names",
            "issues": [
                {
                    "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
                    "fields": {
                        "customfield_10000": "netic.dk"
                    },
                    "id": "285131",
                    "key": "x-6866",
                    "self": "https://jira.netic.dk/rest/api/2/issue/285131"
                },
                {
                    "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
                    "fields": {
                        "customfield_10000": "netic.dk"
                    },
                    "id": "290184",
                    "key": "x-592",
                    "self": "https://jira.netic.dk/rest/api/2/issue/290184"
                }
            ],
            "maxResults": 2,
            "startAt": 0,
            "total": 2494
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************************************************************************************
dc4-1-b6-esx48.curasaas.netic.dk : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
`
Changing the number is reflected in the search results.